src/com/xilinx/rapidwright/edif/EDIFLibrary.java

```diff
@@ -79,16 +79,55 @@ public EDIFCell addCell(EDIFCell cell){
 		return cell;
 	}
 
+	private String findUniqueCellName(String name) {
+		int counter = 0;
+		String counterName = name; //First try without suffix
+		while (cells.containsKey(counterName)) {
+			counter++;
+			counterName = name+"_"+counter;
+		}
+		return counterName;
+	}
+
+	/**
+	 * Adds the provided cell to the library. If a cell by this name already exists, append a suffix to uniqueify the
+	 * name.
+	 * @param cell The cell to add to the library.
+	 * @param preferredSuffix If a suffix needs to be appended for uniquification, use this one
+	 * @return The cell that has been added.
+	 */
+	public EDIFCell addCellRenameDuplicates(EDIFCell cell, String preferredSuffix) {
+		if(cells == null) cells = getNewMap();
+		cell.setLibrary(this);
+
+		EDIFCell collision = cells.put(cell.getName(), cell);
+		if (collision == null) {
+			return cell;
+		}
+		//Restore the old mapping
+		cells.put(cell.getName(), collision);
+
+		if (preferredSuffix == null) {
+			preferredSuffix = "collisionRename";
+		}
+		String newName = findUniqueCellName(cell.getName()+"_HDI_"+preferredSuffix);
```

Comment:
Can you explain this behaviour some more please? The `preferredSuffix` comes from the EDIF rename, and then you're appending it after `_HDI_` to generate a new unique cell name?

Can you give an example?

I'm also thinking that maybe we shouldn't use the `_HDI_` moniker either, since that's what Vivado typically uses.

Reply:
This handles a weird edge case where two cells have the same name but only differ in their rename. This actually occurs in practice. I added an example here: https://github.com/eddieh-xlnx/RapidWrightDCP/pull/12

I'm open for discussion about migrating away from `_HDI_`.

Reply:
I have changed the code to use `_RW_`.